F1WM/Repositories/NewsRepository.cs

```diff
@@ -79,6 +77,65 @@ public async Task<NewsDetails> GetNewsDetails(int id)
 			return news;
 		}
 
+		public async Task<NewsSummaryPaged> GetNewsByTagId(int? tagId, int page, int countPerPage)
+		{
+			await SetDbEncoding();
+			IQueryable<News> dbNews;
+
+			dbNews = context.NewsTagMatch
+					.Where(t => t.TagId == tagId)
+					.Include(t => t.News)
+					.Select(t => t.News)
+					.Where(n => !n.NewsHidden)
+					.Include(n => n.Tag);
+
+			return GetPagedResult(dbNews, page, countPerPage);
+		}
+
+		public async Task<NewsSummaryPaged> GetNewsByTypeId(int? typeId, int page, int countPerPage)
+		{
+			await SetDbEncoding();
+			IQueryable<News> dbNews;
+
+			dbNews = context.News
+				.Where(n => n.TypeId == typeId && !n.NewsHidden)
+				.Include(n => n.Tag);
+
+			return GetPagedResult(dbNews, page, countPerPage);
+		}
+
+		public async Task<IEnumerable<ApiModel.NewsType>> GetNewsTypes()
+		{
+			await SetDbEncoding();
+			IEnumerable<DatabaseModel.NewsType> dbNewsTypes;
+			dbNewsTypes = await context.NewsTypes.ToListAsync();
+			return mapper.Map<IEnumerable<ApiModel.NewsType>>(dbNewsTypes);
+		}
+
+		public async Task<IEnumerable<ApiModel.NewsTag>> GetNewsTags()
+		{
+			await SetDbEncoding();
+			IEnumerable<DatabaseModel.NewsTag> dbNewsTags;
+			dbNewsTags = await context.NewsTopics.ToListAsync();
```

Comment:
No need to split dbNewsTags declaration and assignment. You can just use `var dbNewsTags = ...`